src/V3Options.cpp

```diff
@@ -104,6 +106,223 @@ class V3OptionsImp final {
+            (*this)(optp, act, wrap);
+        }
+
+        // Callback:set(const char*)
+        void operator()(const char* optp, ActCbVal act,
+                        std::function<void(const char*)> cb) const {
+            auto wrap = [cb](const char*, const char* valp) { cb(valp); };
+            m_parser.emplaceToList({TYPE_CB, act.m_act, static_cast<bool*>(nullptr), wrap, optp});
+        }
+        void operator()(const char* optp, ActCbVal act,
+                        std::pair<V3Options*, void (V3Options::*)(const string&)> fp) const {
+            auto wrap = [fp](const char*, const char* valp) { (fp.first->*fp.second)(valp); };
+            m_parser.emplaceToList({TYPE_CB, act.m_act, static_cast<bool*>(nullptr), wrap, optp});
+        }
+        // Callback:partial match wihtout value
+        void operator()(const char* prefixp, ActCbPartialMatch act,
+                        std::function<void(const char*)> cb) const {
+            const int prefixLen = std::strlen(prefixp);
+            auto wrap = [cb, prefixLen](const char* optp, const char*) { cb(optp + prefixLen); };
+            m_parser.emplaceToList(
+                {TYPE_CB, act.m_act, static_cast<bool*>(nullptr), wrap, prefixp});
+        }
+        // Callback:partial match with value
+        void operator()(const char* prefixp, ActCbPartialMatchVal act,
+                        std::function<void(const char*, const char*)> cb) const {
+            const int prefixLen = std::strlen(prefixp);
+            auto wrap = [cb, prefixLen](const char* optp, const char* valp) {
+                cb(optp + prefixLen, valp);
+            };
+            m_parser.emplaceToList(
+                {TYPE_CB, act.m_act, static_cast<bool*>(nullptr), wrap, prefixp});
+        }
+    };
+
+private:
+    std::map<const string, Entry> m_options;
+    VSpellCheck m_spellCheck;
+    Entry* find(const char* optp) {
+        auto it = m_options.find(optp);
+        if (it != m_options.end()) return &it->second;
+        for (auto&& entryPair : m_options) {
+            Entry& entry = entryPair.second;
+            if (entry.m_act == ACT_ONOFF || entry.m_act == ACT_CB_ONOFF) {  // Allow -no
+                const char* const nop = std::strncmp(optp, "-no", 3) ? nullptr : (optp + 3);
+                if (nop && (entry.m_opt == nop || entry.m_opt == (string{"-"} + nop))) {
+                    return &entry;
+                }
+            } else if (entry.m_act == ACT_CB_PARTIAL_MATCH
+                       || entry.m_act == ACT_CB_PARTIAL_MATCH_VAL) {
+                if (!std::strncmp(optp, entry.m_opt.c_str(), entry.m_opt.length())) {
+                    return &entry;
+                }
+            }
+        }
+        return nullptr;
+    }
+    void emplaceToList(const Entry& e) {
+        const string& opt = e.m_opt;
+        UASSERT(opt.size() >= 2, opt << " is too short");
+        UASSERT(opt[0] == '-' || opt[0] == '+', opt << " does not start with either '-' nor '+'");
```

Comment:
```suggestion
        UASSERT(opt[0] == '-' || opt[0] == '+', opt << " does not start with either '-' or '+'");
```
either or, neither nor.